# Factoring

Reduce fractions by factoring. Find common factor between the num and the denom.

The modulus (%) finds the remainder.

In [1]:
print(5%3)

print(31%10)

2
1


Sample output:

$2$

$1$

Use the modulus in a loop to find factors.

In [2]:
number = 12

# Find all factors
for test_factor in range(1, number + 1):
    if number % test_factor == 0:
        print(test_factor)

1
2
3
4
6
12


Sample output:

1<br>
2<br>
3<br>
4<br>
6<br>
12

Reduce fractions to the lowest terms.

In [9]:
numerator = 12
denominator = 24
factor = 1

# Find greatest common factor
for test_factor in range(1, denominator + 1):
    if numerator % test_factor == 0 and denominator % test_factor == 0:
        factor = test_factor

# Divide out greatest common factor
n = int(numerator/factor)
d = int(denominator/factor)

print("original:", numerator, "/", denominator)
print("reduced:", n, "/", d)

original: 12 / 24
reduced: 1 / 2


Output when numerator = 12 and denominator = 24

original: 12 / 24

reduced: 1 / 2

You can add this code to the decimal-to-fraction code

In [13]:
# Get the decimal number to convert
digits = input("Enter a decimal number to convert: ")

# Convert to a fraction
exponent = int(len(digits)) - 1
n = float(digits)
numerator = int(n * 10**exponent)
denominator = 10**exponent

# Reduce that fraction
factor = 1
for test_factor in range(1, denominator + 1):
    if numerator % test_factor == 0 and denominator % test_factor == 0:
        factor = test_factor

# Divide out greatest common factor
num = int(numerator/factor)
den = int(denominator/factor)

# Output
print("The decimal is", n)
print("The fraction is", num, "/", den)

The decimal is 0.125
The fraction is 1 / 8


Some sample outputs:

The decimal is 0.75

The fraction is 3 / 4

<br/>

The decimal is 0.45

The fraction is 9 / 20

<br/>

The decimal is 0.125

The fraction is 1 / 8

Factoring square roots

In [15]:
import math

print(math.sqrt(24))

4.898979485566356


Divide out any perfect square factors. For example:

$\sqrt{12} = \sqrt{4 * 3} = 2\sqrt{3}$

Because 4 is a square number, the square root of 4 is now outside the radical.

In [16]:
import math

# number to factor
n = 12

# This variable will change
max_factor = 1

# the key ingredient
upper_limit = math.floor(math.sqrt(n)) + 1

# Find square factors
for maybe_factor in range(1, upper_limit):
    if n % (maybe_factor **2) == 0:
        max_factor = maybe_factor

# Results so far
print("n =", n)
print("Square rooted factor =", max_factor)
print("Square factor = ", max_factor**2)
print("integer:", n/(max_factor**2))

n = 12
Square rooted factor = 2
Square factor =  4
integer: 3.0


Sample output:

n = 12

Square rooted factor = 2

Square factor =  4

integer: 3.0

Use Sympy to make it look nice

In [25]:
import math
import sympy
from sympy import symbols

n = 55

# Use these variables
upper_limit = math.floor(math.sqrt(n)) + 1
max_factor = 1
other_factor = 1
square_root = 1

# Slightly different variable strategy
for maybe_factor in range(1, upper_limit):
    if n % (maybe_factor ** 2) == 0:
        max_factor = maybe_factor ** 2

# Divide out greatest square factor
other_factor = n/max_factor

# Output variables
square_root = int(math.sqrt(max_factor))
other_factor = int(other_factor)
output = square_root*sympy.sqrt(other_factor)

# Sympy output without print statement - must be last line
output

sqrt(55)

Outputs:

12: $2\sqrt{3}$<br>
24: $2\sqrt{6}$<br>
40: $2\sqrt{10}$<br>